spec/services/course_creation_service_spec.rb

```diff
@@ -0,0 +1,172 @@
+require "rails_helper"
+
+describe CourseCreationService do
+  context "primary course" do
+    subject { described_class.call(course_params: course_params(:primary), provider: provider, next_available_course_code: false) }
+
+    it "create the primary course" do
+      primary_course_params.except("is_send", "sites_ids", "subjects_ids", "course_code").each do |key, value|
+        expect(subject.send(key)).to eq(value)
+      end
+
+      expect(subject.is_send).to eq(true)
+      expect(subject.sites.map(&:id)).to eq([site.id])
+      expect(subject.subjects.map(&:id)).to eq([primary_subject.id])
+      expect(subject.course_code).to eq(nil)
+    end
+
+    context "next_available_course_code is true" do
+      subject { described_class.call(course_params: course_params(:primary), provider: provider, next_available_course_code: true) }
+
+      it "create the primary course" do
+        primary_course_params.except("is_send", "sites_ids", "subjects_ids", "course_code").each do |key, value|
+          expect(subject.send(key)).to eq(value)
```

Comment:
Could we use `.public_send` here for convention (and anywhere else)

Reply:
Done